pkg/controller/clusteringress/resources/route.go

```diff
@@ -40,7 +44,16 @@ func MakeRoutes(ci *networkingv1alpha1.ClusterIngress) ([]*routev1.Route, error)
 	return routes, nil
 }
 
-func makeRoute(ci *networkingv1alpha1.ClusterIngress, host string, index int) (*routev1.Route, error) {
+func makeRoute(ci *networkingv1alpha1.ClusterIngress, host string, index int, rule networkingv1alpha1.ClusterIngressRule) (*routev1.Route, error) {
+	annotations := make(map[string]string)
+
+	for i, _ := range rule.HTTP.Paths {
+		// Supported time units for openshift route annotations are microseconds (us), milliseconds (ms), seconds (s), minutes (m), hours (h), or days (d)
+		// But the timeout value from clusteringress is in xmys(ex: 10m0s) format
+		// So, in order to make openshift route to work converting it into seconds.
+		annotations[routeTimeout] = fmt.Sprintf("%v", rule.HTTP.Paths[i].Timeout.Duration.Seconds()) + "s"
```

Comment:
```suggestion
		annotations[timeoutAnnotation] = fmt.Sprintf("%ds", rule.HTTP.Paths[i].Timeout.Duration.Seconds())
```

Reply:
Sorry my bad i saw your comment and just added while doing PR

Reply:
The return type of Seconds() is float64

So when i use this way
```
fmt.Sprintf("%ds", rule.HTTP.Paths[i].Timeout.Duration.Seconds())
```
The output is as shown below
**haproxy.router.openshift.io/timeout: '%!d(float64=600)s'**

Because %ds needs string value.

Reply:
no it needs a decimal value.

`%vs` should work then. I was mainly getting at the extranous string append.

Reply:
Yes %vs works.